python/pyarrow/dataset.py

```diff
@@ -20,6 +20,16 @@
 import pyarrow as pa
 from pyarrow.util import _stringify_path, _is_path_like
 
+from pyarrow.fs import (
+    _normalize_path,
+    FileSelector,
+    FileSystem,
```

Comment:
Can you import those inside the functions as it was before (or another import that doesn't import the actual names, like `from pyarrow import fs`) ? 

I did that on purpose to avoid adding those names to the namespace. 
(we can discuss adding those here on purpose, but I would keep that for a separate discussion. But we should avoid adding a subset which depends on implementation details)

Reply:
Added `__all__`. I wonder: do we want to expose the low level classes under `pyarrow.dataset`?

Reply:
Adding `__all__` won't help for this I think.

> I wonder: do we want to expose the low level classes under pyarrow.dataset?

Yes I think so, eg kartothek will want to use them

Reply:
Updated.